library/std/src/process.rs

```diff
@@ -1636,8 +1644,10 @@ impl fmt::Display for ExitStatusError {
 #[unstable(feature = "exit_status_error", issue = "84908")]
 impl crate::error::Error for ExitStatusError {}
 
-/// This type represents the status code a process can return to its
-/// parent under normal termination.
+/// This type represents the status code the current process can return
+/// to its parent under normal termination.
```

Comment:
I think this should add an explicit paragraph about being write-only (along these lines):

> ExitCode is intended to be consumed only by the standard library (via `Termination::report()`), and intentionally does not provide accessors like PartialEq/Eq/Hash/...

This will help in the future if someone asks to add PartialEq, since we can know to be a little more careful, but also to help newcomers to avoid digging in to try to figure out how to access the inner code or compare to SUCCESS, etc. -- this tells them there's nothing hidden, it's intentionally not possible.